opendatakit/briefcase
# Issue 246 Pull one form from Collect directory

Closes #246

This PR adds a new optional `--form_id` (`-id` in short form) param to define the ID of the form that the pull from collect CLI operation should pull.

- When the param is defined, the operation pulls only the form matching the given form ID
- Pulls all forms otherwise, like before this PR

#### What has been done to verify that this works as intended?
- Scenario: clean storage directory and a Collect folder with 2 forms `choice-lists` and `choice-lists-dynamic`
- Pull from collect without `--form_id`. Verify with the UI that now there are the 2 forms
- Start again
- Pull from collect with `--form_id choice-lists`. Verify with the UI that now there is just the form `choice-lists`
- Start again
- Pull from collect with `--form_id choice-lists-dynamic`. Verify with the UI that now there is just the form `choice-lists-dynamic`

I used this Collect directory: [odk.zip](https://github.com/opendatakit/briefcase/files/2179192/odk.zip)

#### Why is this the best possible solution? Were any other approaches considered?
This is a straightforward change that imitates the same filtering we do in other pull operations that use the same CLI param

#### Are there any risks to merging this code? If so, what are they?
Nope.

#### Does this change require updates to documentation? If so, please file an issue at https://github.com/opendatakit/docs/issues/new and include the link below.
Nope.
